[Page][Olana] Hackathons List

jrolana opened the pull request:
Note:
Created only dummy data for currently log in user will update after finishing user register/sign in page.

jrolana committed:
[Page][Olana] Hackathons List

Note:
Created only dummy data for currently log in user will update after finishing user register/sign in page.

ezerssss commented on hackathons_list.php:
```diff
@@ -0,0 +1,113 @@
+<head>
+    <meta charset="UTF-8">
+    <meta name="viewport" content="width=device-width, initial-scale=1.0">
+    <title>Document</title>
```
change title

ezerssss commented on hackathons_list.php:
```diff
@@ -0,0 +1,113 @@
+<body>
+    <?php
+    include 'db_connector.php';
+    $currentUserID = 2;
```
from session to right? pero like question lang ba para like on the same page di siya correcting or somethni GHAHAHAHA

jrolana replied:
yes for session

ezerssss commented on hackathons_list.php:
```diff
@@ -0,0 +1,113 @@
+            while ($joinedHackathon = $joined->fetch_assoc()) {
+                $joinedHackathonID = $joinedHackathon["hackathonID"];
+                $hackathon_sql = "SELECT * FROM hackathons WHERE hackathonID=$joinedHackathonID";
+                $hackathon_details = $conn->query($hackathon_sql);
```
can probably make this into a simpler one SQL query

```
SELECT * FROM hackathons WHERE hackathonsID IN (SELECT hackathonID FROM participates WHERE userID=currentUserID AND dateEnd > $currentDate)
```

maybe a little bit readable is:
```
$joinedHackathons_sql = SELECT hackathonID FROM participates WHERE userID=currentUserID AND dateEnd > $currentDate;

$hackathons = SELECT * FROM hackathons WHERE hackathonsID IN ($joinedHackathons_sql)
```

ezerssss commented on hackathons_list.php:
```diff
@@ -0,0 +1,113 @@
+                                "<h3>" . $hackathon["theme"] . "</h3>" .
+                                "<p>" . $hackathon["description"] . "</p>" .
+                                "<p>" . $hackathon["dateStart"] . " to " . $hackathon["dateEnd"] . "</p>" .
+                                "<button>Submit a project</button>" .
```
change to a form lang para ready na siya for integration soon

ezerssss commented on hackathons_list.php:
```diff
@@ -0,0 +1,113 @@
+
+        if ($available) {
+            while ($availHackathon = $available->fetch_assoc()) {
+                $availHackathonID = $availHackathon["hackathonID"];
```
i think same sa previous ko na comment nga better if

```
$joinedHackathons_sql = SELECT hackathonID FROM participates WHERE userID=currentUserID;

$hackathons = SELECT * FROM hackathons WHERE hackathonsID NOT IN ($joinedHackathons_sql)
```

ezerssss commented on hackathons_list.php:
```diff
@@ -0,0 +1,113 @@
+                                "<h3>" . $hackathon["theme"] . "</h3>" .
+                                "<p>" . $hackathon["description"] . "</p>" .
+                                "<p>" . $hackathon["dateStart"] . " to " . $hackathon["dateEnd"] . "</p>" .
+                                "<button>Join</button>" .
```
make into form

ezerssss commented on scratch.txt:
delete non final files in PR

ezerssss commented on hackathons_list.php:
```diff
@@ -0,0 +1,113 @@
+
+                if ($hackathon_details) {
+                    while ($hackathon = $hackathon_details->fetch_assoc()) {
+                        if ($hackathon["dateEnd"] < $currentDate) {
```
ohhh here mo linagay ang if statement, better siguro if sa actual SQL ang checking para smaller rows ang makuha niya

```
$pastJoinedHackathons_sql = SELECT hackathonID FROM participates WHERE userID=currentUserID AND dateEnd <= $currentDate;

$hackathons = SELECT * FROM hackathons WHERE hackathonsID IN ($joinedHackathons_sql)
```

jrolana committed:
[Fix][Olana] Hackathons list page issue

- Simplified SQL queries
- Modified conditions for joined, available, and past hackathons

jrolana committed:
Merge branch 'main' into olana_page_hackathons_list
